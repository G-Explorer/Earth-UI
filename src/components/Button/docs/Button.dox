/**
 * @title 默认按钮
 */
import Button from 'earth-ui/lib/Button'

const ButtonDefault = () => {
  return <Button>Default Button</Button>
}

/**
 * @title 主次按钮
 */
import Button from 'earth-ui/lib/Button'

const ButtonPST = () => {
  return (
    <div>
      <Button type="primary">Primary</Button>
      <Button type="secondary">Secondary</Button>
      <Button type="tertiary">Tertiary</Button>
    </div>
  )
}

/**
 * @title 颜色按钮
 */
import Button from 'earth-ui/lib/Button'

const ButtonColor = () => {
  return (
    <div>
      <Button type="accept">Accept</Button>
      <Button type="warning">warning</Button>
      <Button type="danger">Danger</Button>
    </div>
  )
}

/**
 * @title 微型按钮
 */
import Button from 'earth-ui/lib/Button'

const ButtonMicro = () => {
  return (
    <div>
      <Button type="primary" micro>Primary</Button>
      <Button type="secondary" micro>Secondary</Button>
      <Button type="tertiary" micro>Tertiary</Button>
    </div>
  )
}

/**
 * @title Block 按钮
 * @desc block属性将使按钮适合其父宽度
 */
import Button from 'earth-ui/lib/Button'

const ButtonBlock = () => {
  return (
    <div>
      <Button type="primary" block>Primary</Button>
      <Button type="secondary" block>Secondary</Button>
      <Button type="tertiary" block>Tertiary</Button>
    </div>
  )
}

/**
 * @title 图标按钮
 * @desc 如果需要自定义图标位置，结合使用 Icon 组件
 */
import Button from 'earth-ui/lib/Button'

const ButtonIcon = () => {
  return (
    <div>
      <Button icon="delete" />
      <Button icon="statuschange" type="primary" />
      <Button icon="statuschange" type="secondary" />
      <Button icon="activity" type="secondary">数据分析</Button>
      <Button icon="statuschange" type="tertiary" />
    </div>
  )
}

/**
 * @title 不可用的
 */
import Button from 'earth-ui/lib/Button'

const ButtonDisabled = () => {
  return (
    <div>
      <Button disabled>不可用</Button>
      <Button icon="statuschange" disabled />
      <Button icon="activity" type="secondary" disabled>数据分析</Button>
    </div>
  )
}

@component Button
