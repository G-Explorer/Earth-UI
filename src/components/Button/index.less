@import '~ui-variables';
@import '~ui-mixins';

.ui-mixin-keyframes(
  move-progress-bar,
  {background-position: 0 0;},
  {background-position: @ui-unit-double @ui-unit-double;}
);

.@{prefix-cls}-button {
  .ui-mixin-unselectable();
  cursor: pointer;
  max-width: 100%;
  margin-right: @ui-unit-half;
  align-items: center;
  position: relative;
  display: inline-flex;
  flex-direction: column;
  justify-content: center;
  transition: background-color .2s ease, color .2s ease, border-color .2s ease;

  &:not(.@{prefix-cls}-button--text):not(.@{prefix-cls}-button--link) {
    &:hover {
      span {
        text-decoration: none;
      }
    }
    span {
      .ui-mixin-button-text();
      .ui-mixin-text-ellipsis();
      padding: 0 @ui-unit-half;
      white-space: nowrap;
      word-wrap: normal;
      overflow: hidden;
    }
  }

  &:last-child {
    margin-right: 0;
  }

  & i.@{prefix-cls}-icon {
    .ui-mixin-reduce-l();
    margin-right: 0;
    font-size: 20px;
    line-height: @ui-fontsize;
    vertical-align: middle;
  }

  &--block {
    width: 100%;
    margin-top: @ui-unit-half;
    &:first-child {
      margin-top: 0;
    }
  }
}

// Type.........................................................................

.@{prefix-cls}-button {
  color: @ui-color-black;
  border: 1px solid @ui-color-gray-light;
  background-color: fade(@ui-color-white, 50%);
  &:hover {
    background-color: fade(@ui-color-gray-light, 50%);
  }
  &:visited {
    background-color: fade(@ui-color-white, 50%);
  }

  &--primary {
    color: @theme-text;
    border: 1px solid @theme;
    background-color: @theme;
    &:hover {
      border-color: @theme-dark;
      background-color: @theme-dark;
    }
  }

  &--accept {
    color: @ui-color-white;
    border: 1px solid @ui-color-green;
    background-color: @ui-color-green;
    &:hover {
      border-color: @ui-color-green-dark;
      background-color: @ui-color-green-dark;
    }
  }

  &--warning {
    color: @ui-color-white;
    border: 1px solid @ui-color-orange;
    background-color: @ui-color-orange;
    &:hover {
      border-color: @ui-color-orange-dark;
      background-color: @ui-color-orange-dark;
    }
  }

  &--danger {
    color: @ui-color-white;
    border: 1px solid @ui-color-red;
    background-color: @ui-color-red;
    &:hover {
      border-color: @ui-color-red-dark;
      background-color: @ui-color-red-dark;
    }
  }

  &--link, &--text {
    color: @theme;
    background-color: transparent !important;
    border: none !important;
    margin-left: @ui-unit-half !important;
    i.@{prefix-cls}-icon {
      font-size: @ui-fontsize;
      padding-left: 0;
    }
    div i {
      padding-right: @ui-unit-quarter;
    }
  }
  &--link:first-child, &--text:first-child {
      margin-left: 0 !important;
  }
  &--link:hover {
    color: @theme-dark;
    span {
      text-decoration: underline;
    }
  }
  &--text:hover {
    color: @theme-dark;
  }
}


// Ghost Button.................................................................

.@{prefix-cls}-button--ghost.@{prefix-cls}-button {
  color: @ui-color-black;
  border: 1px solid @ui-color-gray-light;
  background-color: transparent;
  &:hover {
    background-color: fade(@ui-color-gray-light, 50%);
  }
  &--primary {
    border: 1px solid @theme;
    color: @theme;
    &:hover {
      color: @theme;
      background-color: fade(@theme, 20%);
    }
  }
  &--accept {
    border: 1px solid @ui-color-green;
    color: @ui-color-green;
    &:hover {
      color: @ui-color-green;
      background-color: fade(@ui-color-green, 20%);;
    }
  }
  &--warning {
    border: 1px solid @ui-color-orange;
    color: @ui-color-orange;
    &:hover {
      color: @ui-color-orange;
      background-color: fade(@ui-color-orange, 20%);;
    }
  }
  &--danger {
    border: 1px solid @ui-color-red;
    color: @ui-color-red;
    &:hover {
      color: @ui-color-red;
      background-color: fade(@ui-color-red, 20%);;
    }
  }
  &--link, &--text {
    color: @theme;
  }
  &--link:hover, &--text:hover {
    color: @theme-dark;
  }
}

// Loading bar..................................................................

.@{prefix-cls}-button--loading {
  &,
  &:hover {
    background-color: transparent !important;
    border-color: transparent !important;
    cursor: default !important;
  }
  & > * {
    visibility: hidden;
  }
  &:after {
    content: '';
    background-size: @ui-unit-double @ui-unit-double;
    border-radius: inherit;
    visibility: visible;
    .ui-mixin-maximize();
    .ui-mixin-progress-bar-gradient(
            fade(@ui-color-gray-light, 65%),
            fade(@ui-color-gray-lighter, 65%)
    );
    .ui-mixin-animation(move-progress-bar, 1s, linear, infinite);
    margin: @ui-unit-half;
  }
  &.@{prefix-cls}-button--sm {
    &:after {
      zoom: .36;
    }
  }
}

// Size.........................................................................

.@{prefix-cls}-button--sm {
  &:not(.@{prefix-cls}-button--text):not(.@{prefix-cls}-button--link) {
    .ui-mixin-fixheight(@ui-unit);
    border-radius: @ui-unit-half;
    padding: 0;
    font-size: @ui-fontsize-small;
    min-width: @ui-unit;
  }
  &:not(:disabled):after {
    .ui-mixin-maximize(0-@ui-unit-half,0,0-@ui-unit-half,0);
  }
  i {
    font-size: @ui-fontsize-medium;
    &:before {
      font-size: @ui-fontsize-medium;
    }
  }
}

.@{prefix-cls}-button--md {
  &:not(.@{prefix-cls}-button--text):not(.@{prefix-cls}-button--link) {
    min-width: 82px;
    padding: @ui-unit-half - 1px;
    border-radius: @ui-unit;
  }
}

.@{prefix-cls}-button--lg {
  margin-right: @ui-unit;
  &:not(.@{prefix-cls}-button--text):not(.@{prefix-cls}-button--link) {
    .ui-mixin-fixheight(@ui-unit-plus);
    border-radius: 28px;
    padding: 24px;
    font-size: @ui-fontsize-big;
    min-width: @ui-unit-plus;
  }
}

// Colorful background..........................................................

.@{prefix-cls}-bg-colorful .@{prefix-cls}-button {
  color: @ui-color-white;
  border: 1px solid @ui-color-white;
  background-color: transparent;
  &:hover {
    color: @ui-color-white;
    border: 1px solid fade(@ui-color-white, 30%);
    background-color: fade(@ui-color-white, 30%);
  }
  &--primary {
    color: @ui-color-black;
    background-color: @ui-color-white;
    &:hover {
      color: fade(@ui-color-black, 70%);
      background-color: @ui-color-white;
    }
  }
  &--accept {
    color: @ui-color-green;
    background-color: @ui-color-white;
    &:hover {
      color: fade(@ui-color-green, 70%);
      border: 1px solid @ui-color-white;
      background-color: @ui-color-white;
    }
  }
  &--warning {
    color: @ui-color-orange;
    background-color: @ui-color-white;
    &:hover {
      color: fade(@ui-color-orange, 70%);
      border: 1px solid @ui-color-white;
      background-color: @ui-color-white;
    }
  }
  &--danger {
    color: @ui-color-red;
    background-color: @ui-color-white;
    &:hover {
      color: fade(@ui-color-red, 70%);
      border: 1px solid @ui-color-white;
      background-color: @ui-color-white;
    }
  }
  &--link, &--text {
    color: @ui-color-white;
    &:hover {
      color: fade(@ui-color-white, 70%);
    }
  }
  &[disabled] {
    background-color: fade(@ui-color-white, 30%);
    border: 1px solid fade(@ui-color-white, 0%);
    color: fade(@ui-color-white, 40%);
    &:hover {
      background-color: fade(@ui-color-white, 30%);
      border: 1px solid fade(@ui-color-white, 0%);
      color: fade(@ui-color-white, 40%);
    }
  }
}

// Icon Button..................................................................

.@{prefix-cls}-button__icon-only {
  &:not(.@{prefix-cls}-button--text):not(.@{prefix-cls}-button--link) {
    width: @ui-unit-double;
    height: @ui-unit-double;
    min-width: @ui-unit;
    border-radius: 50%;
  }
  &.@{prefix-cls}-button--sm {
    width: @ui-unit !important;
    i.@{prefix-cls}-icon:before {
      font-size: 13px;
    }
  }
  i.@{prefix-cls}-icon {
    .ui-mixin-reduce-l(0);
    font-size: @ui-fontsize-icon;
  }
}

// Disabled ....................................................................

.@{prefix-cls}-button[disabled] {
  cursor: not-allowed;
  color: fade(@ui-color-gray, 80%);
  border: 1px solid @ui-color-gray-lighter;
  background-color: @ui-color-gray-lighter;
  &:hover {
    color: fade(@ui-color-gray, 80%);
    border: 1px solid @ui-color-gray-lighter;
    background-color: @ui-color-gray-lighter;
  }
}

.@{prefix-cls}-button--link[disabled], .@{prefix-cls}-button--text[disabled] {
  margin-left: @ui-unit-half;
  &:first-child {
    margin-left: 0;
  }
}
