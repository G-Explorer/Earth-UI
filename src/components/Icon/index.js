import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import './index.less'

const Icon = props => {
  const { className, type, src, ...other } = props
  return (
    <i className={cx(`${prefixCls}-icon`, {[`${prefixCls}-icon__${type}`]: type}, className)} {...other}>
      {src && <svg><use xlinkHref={src} /></svg>}
    </i>
  )
}

Icon.propTypes = {
  className: PropTypes.string,
  // 图标类型
  type: PropTypes.string
}

export default Icon
