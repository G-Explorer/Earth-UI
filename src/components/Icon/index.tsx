import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './index.less';

const Icon = (props) => {
    const { className, type, src, ...other } = props;
    return (
        <i
            className={cx(`${prefixCls}-icon`, { [`${prefixCls}-icon__${type}`]: type }, className)}
            {...other}
        >
            {src && (
                <svg>
                    <use xlinkHref={src} />
                </svg>
            )}
        </i>
    );
};

Icon.propTypes = {
    className: PropTypes.string,
    /** icon type */
    type: PropTypes.string,
    /** icon src, such as svg `xlinkHref` */
    src: PropTypes.string,
};

export default Icon;
