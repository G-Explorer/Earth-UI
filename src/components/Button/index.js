import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import Icon from '../Icon'
import './index.less'

class Button extends React.Component {
  constructor (props) {
    super(props)
    this.button = React.createRef()
  }

  componentDidMount () {
    this.props.autoFocus && this.focus()
  }

  componentDidUpdate () {
    this.props.autoFocus && this.focus()
  }

  focus = () => {
    window.setTimeout(() => {
      this.button.current.focus()
    }, 0)
  }

  render () {
    const {
      children,
      className,
      type,
      loading,
      micro,
      block,
      icon,
      autoFocus,
      focus,
      disabled,
      ...other
    } = this.props
    const classNames = cx(
      `${prefixCls}-button`,
      {
        [`${prefixCls}-button_${type}`]: type,
        [`${prefixCls}-button--loading`]: loading,
        [`${prefixCls}-button_micro`]: micro,
        [`${prefixCls}-button_block`]: block,
        [`${prefixCls}-button_nofocus`]: !focus && !autoFocus,
        [`${prefixCls}-button__icon-only`]: icon && !children
      },
      className
    )
    return (
      <button
        type="button"
        className={classNames}
        ref={this.button}
        onClick={this.handleClick}
        disabled={disabled || loading}
        {...other}
      >
        {icon && <Icon type={icon} />}
        {children && <span>{children}</span>}
      </button>
    )
  }
}

Button.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,

  // 按钮类型
  type: PropTypes.oneOf(['primary', 'secondary', 'tertiary', 'accept', 'warning', 'danger']),

  // 是否为微型按钮
  micro: PropTypes.bool,

  // 是否为block按钮
  block: PropTypes.bool,

  // Initialize a spinner
  loading: PropTypes.bool,

  // Disabled button
  disabled: PropTypes.bool,

  // 按钮图标
  icon: PropTypes.string,

  // 按钮是否带 focus 的效果，默认 `false`
  focus: PropTypes.bool,

  // 按钮是否自动为 focus 状态，默认 `false`
  autoFocus: PropTypes.bool
}

Button.defaultProps = {
  focus: false,
  autoFocus: false
}

export default Button
