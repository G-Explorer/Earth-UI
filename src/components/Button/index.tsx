import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Icon from '../Icon';
import './index.less';

function Button(props) {
    const {
        children,
        className,
        type,
        loading,
        size,
        block,
        ghost,
        icon,
        disabled,
        htmlType,
        ...other
    } = props;
    const classNames = cx(
        `${prefixCls}-button`,
        {
            [`${prefixCls}-button--${type}`]: type,
            [`${prefixCls}-button--${size}`]: size,
            [`${prefixCls}-button--ghost`]: ghost,
            [`${prefixCls}-button--loading`]: loading,
            [`${prefixCls}-button--block`]: block,
            [`${prefixCls}-button__icon-only`]: icon && !children,
        },
        className
    );
    if (icon && children) {
        return (
            <button
                type={htmlType}
                className={classNames}
                disabled={disabled || loading}
                {...other}
            >
                <div>
                    <Icon type={icon} />
                    <span>{children}</span>
                </div>
            </button>
        );
    }
    return (
        <button type={htmlType} className={classNames} disabled={disabled || loading} {...other}>
            {icon && <Icon type={icon} />}
            {children && <span>{children}</span>}
        </button>
    );
}

Button.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,

    /** button type which is not original html `type` of `button` */
    type: PropTypes.oneOf(['primary', 'accept', 'warning', 'danger', 'link', 'text']),

    /** button size */
    size: PropTypes.oneOf(['sm', 'md', 'lg']),

    /** determine whether it's a ghost button */
    ghost: PropTypes.bool,

    /** determine whether it's a block button */
    block: PropTypes.bool,

    /** determine whether it's loading */
    loading: PropTypes.bool,

    /** disable the button */
    disabled: PropTypes.bool,

    /** icon type name */
    icon: PropTypes.string,

    /** set the original html `type` of `button`, see: [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type) */
    htmlType: PropTypes.oneOf(['button', 'submit', 'reset']),
};

Button.defaultProps = {
    size: 'md',
    htmlType: 'button',
};

export default Button;
