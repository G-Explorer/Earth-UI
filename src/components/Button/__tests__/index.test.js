import React from 'react'
import { render, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import Button from '../index'
import Icon from '../../Icon'

describe('Button', () => {
    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {})

    afterEach(() => {
        errorSpy.mockReset()
    })

    afterAll(() => {
        errorSpy.mockRestore()
    })

    it('should render correctly', () => {
        const wrapper = render(<Button>OK</Button>)
        expect(wrapper).toMatchSnapshot()
    })
    it('should mount correctly', () => {
        expect(() => renderer.create(<Button>OK</Button>)).not.toThrow()
    })
    it('should render className correctly', () => {
        const wrapper = mount(<Button className="test">OK</Button>)
        expect(wrapper.find('button').hasClass('test')).toBe(true)
    })
    it('should render style correctly', () => {
        const wrapper = render(<Button style={{ color: 'red' }}>OK</Button>)
        expect(wrapper.prop('style').color).toBe('red')
    })
    it('should render Chinese characters correctly', () => {
        const wrapper = render(<Button>添加</Button>)
        expect(wrapper).toMatchSnapshot()
    })
    it('should render icon correctly', () => {
        const wrapper1 = render(<Button icon="add">添加</Button>)
        expect(wrapper1).toMatchSnapshot()
        const wrapper2 = render(
            <Button>
                <Icon type="add" />
                添加
            </Button>
        )
        expect(wrapper2).toMatchSnapshot()
    })
    it('should trigger event function correctly', () => {
        const handleClick = jest.fn()
        const wrapper = mount(<Button onClick={handleClick}>OK</Button>)
        wrapper.simulate('click')
        expect(handleClick).toBeCalled()
        expect(handleClick.mock.calls.length).toBe(1)
    })
    it('should render disabled button correctly', () => {
        const wrapper = render(<Button disabled>OK</Button>)
        expect(wrapper.prop('disabled')).toBe(true)
    })
    it('should render loading button correctly', () => {
        const wrapper = mount(
            <Button type="primary" loading>
                Show Loading
            </Button>
        )
        expect(wrapper.find('button').hasClass('true-button--loading')).toBe(true)
    })
    it('should not trigger event function when button is loading', () => {
        const handleClick = jest.fn()
        const wrapper = mount(
            <Button onClick={handleClick} loading>
                OK
            </Button>
        )
        wrapper.simulate('click')
        expect(handleClick).not.toBeCalled()
    })
    it('should render type correctly', () => {
        const types = ['primary', 'accept', 'warning', 'danger', 'text', 'link']
        const wrappers = [
            mount(<Button type={types[0]}>PRIMARY</Button>),
            mount(<Button type={types[1]}>ACCEPT</Button>),
            mount(<Button type={types[2]}>WARNING</Button>),
            mount(<Button type={types[3]}>DANGER</Button>),
            mount(<Button type={types[4]}>TEXT</Button>),
            mount(<Button type={types[5]}>LINK</Button>)
        ]
        wrappers.forEach((wrapper, index) => {
            expect(wrapper.find('button').hasClass(`true-button--${types[index]}`)).toBe(true)
        })
        const wrapperDefault = mount(<Button>DEFAULT</Button>)
        types.forEach(type => {
            expect(wrapperDefault.find('button').hasClass(`true-button--${type}`)).toBe(false)
        })
    })
    it('should render size correctly according to its size prop', () => {
        const wrapperSm = mount(<Button size="sm">OK</Button>)
        const wrapperMd = mount(<Button size="md">OK</Button>)
        const wrapperLg = mount(<Button size="lg">OK</Button>)
        expect(wrapperSm.find('button').hasClass('true-button--sm')).toBe(true)
        expect(wrapperMd.find('button').hasClass('true-button--md')).toBe(true)
        expect(wrapperLg.find('button').hasClass('true-button--lg')).toBe(true)
    })
    it('should render middle size without size prop', () => {
        const wrapperDefault = mount(<Button>OK</Button>)
        expect(wrapperDefault.find('button').hasClass('true-button--md')).toBe(true)
    })
    it('should render ghost button correctly', () => {
        const wrappers = [
            mount(<Button ghost>DEFAULT</Button>),
            mount(
                <Button ghost type="primary">
                    PRIMARY
                </Button>
            ),
            mount(
                <Button ghost type="accept">
                    ACCEPT
                </Button>
            ),
            mount(
                <Button ghost type="warning">
                    WARNING
                </Button>
            ),
            mount(
                <Button ghost type="danger">
                    DANGER
                </Button>
            ),
            mount(
                <Button ghost type="text">
                    TEXT
                </Button>
            ),
            mount(
                <Button ghost type="link">
                    LINK
                </Button>
            )
        ]
        wrappers.forEach(wrapper => {
            expect(wrapper.find('button').hasClass('true-button--ghost')).toBe(true)
        })
    })
    it('should render original html type of button correctly', () => {
        const wrapperDefault = mount(<Button>Default</Button>)
        const wrapperSubmit = mount(<Button htmlType="submit">Default</Button>)
        const wrapperReset = mount(<Button htmlType="reset">Reset</Button>)
        expect(wrapperDefault.find('button').prop('type')).toBe('button')
        expect(wrapperSubmit.find('button').prop('type')).toBe('submit')
        expect(wrapperReset.find('button').prop('type')).toBe('reset')
    })
    it('should have error when setting invalid htmlType', () => {
        mount(<Button htmlType="hello">Hello</Button>)
        expect(errorSpy).toHaveBeenCalled()
    })
})
