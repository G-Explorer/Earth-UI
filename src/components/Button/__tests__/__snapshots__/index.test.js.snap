// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Chinese characters correctly 1`] = `
<button
  class="true-button true-button_nofocus"
  type="button"
>
  <span>
    添加
  </span>
</button>
`;

exports[`Button should render correctly 1`] = `
<button
  class="true-button true-button_nofocus"
  type="button"
>
  <span>
    OK
  </span>
</button>
`;

exports[`Button should render icon correctly 1`] = `
<button
  class="true-button true-button_nofocus"
  type="button"
>
  <i
    class="true-icon true-icon__add"
  />
  <span>
    添加
  </span>
</button>
`;

exports[`Button should render icon correctly 2`] = `
<button
  class="true-button true-button_nofocus"
  type="button"
>
  <span>
    <i
      class="true-icon true-icon__add"
    />
    添加
  </span>
</button>
`;
